### DISCUSSION TITLE
SunnyTech ☀️

### DISCUSSION THREAD

**Discussion by mickaelalvs** [lyonjs/shortvid.io#111] [Conference / Meetup Communication]
Let's see what we can do to help "SunnyTech" conference 👋🏼

**Comment by HugoGresse** [comment]
Hey,

As part of our initiatives at SunnyTech 2024, we are advancing towards automating the speaker announcement process. Our speaker management is facilitated through [OpenPlanner.fr](https://github.com/HugoGresse/OpenPlanner), which acts as a comprehensive CRUD interface for our entire website and numerous other events.

The plan involves creating a feature that allows us to generate videos for all talks with a single click and then utilize the unofficial Buffer API to schedule these posts as drafts.

Upon reviewing the ShortVid website, the API call looks like this:

```javascript
await fetch("https://api.shortvid.io/Talk", {
    "referrer": "https://shortvid.io/",
    "body": "{\"talkTitle\":\"Example\",\"speakersNames\":\"John Doe\",\"titleSize\":\"50\",\"eventLogo\":\"\"}",
    "method": "POST"
});
```


What should be your guidelines for implementing this? Are you okay with us integrating ShortVid.io into OpenPlanner for everyone to use? What would be the associated costs?

**Reply by Slashgear** [reply]
With @CruuzAzul we do not know the existence of OpenPlanner, what a nice project, we could be really interessed on a demo of it.
This may help us for LyonJS meetup, we never know.

Yes you successfully reverse engineered our API to generate video.

This should work in your want to generate the video on the backend side of OpenPlanner.(We have a very closed CORS policy on it, no client side call possible)

Please pass a referrer to the query to let us understand who is generating video.

As your solution seems to be free, we let you generate video for it.
Hosting cost is offered by our sponsors so you don't need to worry too much (but if you generate too much video we may need to put limitation in the futur, rate limiting, auth on API)

**Reply by HugoGresse** [reply]
Based on the [API source](https://github.com/lyonjs/shortvid.io/blob/main/server.mjs)  

Wouldn't it be better to have our own serverless function as part of OpenPlanner (while still mentionning the Shortvid)? I mean it would not impact Shortvid ressources then. I dunno about the cost assotiated though, Firebase Functions may be pricier or not, though adding ffmpeg and other native dep may not be very easy. 

Or we could move to that in the future depending on the usage.

**Reply by Slashgear** [reply]
Today, only stable solution to host on Functions is an AWS lambda and they are not currenlty our sponsor ^^'
